lib/bandit/http1/handler.ex

```diff
@@ -140,9 +140,16 @@ defmodule Bandit.HTTP1.Handler do
 
     if under_limit && req.keepalive do
       case ensure_body_read(req) do
-        :ok -> {:continue, Map.put(state, :requests_processed, requests_processed)}
-        {:error, :closed} -> {:close, state}
-        {:error, reason} -> {:error, reason, state}
+        :ok ->
+          gc_every_n_requests = Keyword.get(state.opts.http_1, :gc_every_n_keepalive_requests, 5)
+          if rem(requests_processed, gc_every_n_requests) == 0, do: :erlang.garbage_collect()
+          {:continue, Map.put(state, :requests_processed, requests_processed)}
```

Comment:
__Inner monologue:__ "Could we reproduce, via test, the memory bloat described in #313 with a sequence of keepalive requests over a single connection? And then assert that the memory is reduced about N of those requests?"

Reply:
☝️ To be clear, not implying you need a test before merge. Just wondered if the above would be possible.

Reply:
I think it'd have to be in an `async: false` test to isolate things, but maybe even then it wouldn't be possible?

Reply:
Yeah...... I was thinking the same thing. I can likely dig into the Erlang statistics call and validate. lemme look into it

Reply:
> there's no explicit way to get a count / notice of full sweep GCs that I can find, so I've gotta resort to side effects

I think that's better, as the side effect is the important thing, right?

And if you can reliably red/green it by changing the config, 🤌. Great work!